stripe/src/main/java/com/stripe/android/CustomerSession.java

```diff
@@ -246,8 +259,12 @@ public void addCustomerSource(
         final Map<String, Object> arguments = new HashMap<>();
         arguments.put(KEY_SOURCE, sourceId);
         arguments.put(KEY_SOURCE_TYPE, sourceType);
-        mSourceRetrievalListener = listener;
-        mEphemeralKeyManager.retrieveEphemeralKey(ACTION_ADD_SOURCE, arguments);
+
+        final String operationId = UUID.randomUUID().toString();
+        if (listener != null) {
+            mSourceRetrievalListenerRefs.put(operationId, new WeakReference<>(listener));
+        }
+        mEphemeralKeyManager.retrieveEphemeralKey(operationId, ACTION_ADD_SOURCE, arguments);
```

Comment:
Is the `operationId` needed if the listener is null?

Reply:
see below